controllers/v1/observationsController.js

```diff
@@ -982,42 +993,67 @@ module.exports = class Observations extends Abstract {
                     result: {}
                 };
 
+                //eG change
                 let observationDocument = await database.models.observations.findOne({ 
                     _id: req.params._id, 
                      createdBy: req.userDetails.userId,
                      status: {$ne:"inactive"}, 
-                     entities: ObjectId(req.query.entityId) }).lean();
+                     entities: req.query.entityId }).lean();
 
                 if (!observationDocument) {
                     return resolve({ 
                         status: httpStatusCode.bad_request.status, 
                         message: messageConstants.apiResponses.OBSERVATION_NOT_FOUND
                     });
                 }
-               
-                let entityQueryObject = { _id: req.query.entityId, entityType: observationDocument.entityType };
-                let entityDocument = await database.models.entities.findOne(
-                    entityQueryObject,
-                    {
-                        metaInformation: 1,
-                        entityTypeId: 1,
-                        entityType: 1,
-                        registryDetails: 1
+               //eG code
+                let entityResult = [];
+                let filterData = {
+                    "id" : req.query.entityId
+                };
+
+                let entitiesDocument = await sunbirdService.learnerLocationSearch( filterData );
+
+                if ( !entitiesDocument.success || !entitiesDocument.data.response > 0 ) {
+                    let responseMessage = messageConstants.apiResponses.ENTITY_NOT_FOUND;
+                    return resolve({ 
+                        status: httpStatusCode.bad_request.status, 
+                        message: responseMessage 
+                    });
+                }
+                let entities = entitiesDocument.data.response;
+                entities.map(entity => {
+                    if( entity.type == observationDocument.entityType ) {
+                        entityResult.push(entity);
                     }
-                ).lean();
+                });  
 
-                if (!entityDocument) {
+                if ( !entityResult.length > 0 ) {
                     let responseMessage = messageConstants.apiResponses.ENTITY_NOT_FOUND;
                     return resolve({ 
                         status: httpStatusCode.bad_request.status, 
                         message: responseMessage 
                     });
                 }
+                
+                let entityDocument = {
```

Comment:
@VISHNUDAS-tunerlabs make it a common function "extractDataFromLocationResult" from extracting these values. on top also the same code is repeating.